### PR TITLE
Add Python 3.8 testing to Travis

### PR DESCRIPTION
**Pull request by asmeurer**
#### References to other Issues or PRs

See https://github.com/sympy/sympy/pull/17005

#### Release Notes

NO ENTRY

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,7 +44,22 @@ matrix:
...
+      python: 3.8-dev
+      dist: xenial
+      sudo: true
+      env:
+        - TEST_DOCTESTS="true" FASTCACHE="false" TEST_SETUP="true"
+    - python: 3.8-dev
+      dist: xenial
+      sudo: true
+      env:
+        - SPLIT="2/2" TEST_SYMPY="true"
```
Comment (oscargus) [inline .travis.yml]: 1/2

**Comment by oscargus** [comment]
Looks good! (At least once the split issue is solved.)

File: .travis.yml (hunk 1 above)
Comment (asmeurer) [reply .travis.yml]: Good catch. I don't know how that typo happened.

**Comment by asmeurer** [comment]
Looks like the compilation tests fail with ` /usr/bin/ld: cannot find -lpython3.8`

**Comment by asmeurer** [comment]
I'm a little confused why the 3.8 virtualenv on Travis has Cython installed in it. I know Cython needed some fixes for 3.8 so maybe they included it for compatibility.

**Comment by asmeurer** [comment]
There's a warning in the doctests

```
<doctest sympy.core.singleton.SingletonRegistry[3]>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
  0 is S.Zero
```

We should make it so that warnings in the doctests raise an exception, rather than just printing to the terminal.

**Comment by asmeurer** [comment]
Reopened https://github.com/sympy/sympy/issues/16977 for that. Let's merge this for now since it doesn't produce a failure, so that we have 3.8 testing on Travis.

**Pull request merged**